src/metabase/driver/sql/query_processor.clj

```diff
@@ -109,6 +121,17 @@
   [driver]
   :%now)
 
+(defmulti current-datetime-base-type
+  "Field :base-type that is returned by the DB specific \"current datetime\" HSQL form
+  (i.e. `current-datetime-honeysql-form`).  Defaults to `:type/DateTime`."
+  {:arglists '([driver])}
+  driver/dispatch-on-initialized-driver
+  :hierarchy #'driver/hierarchy)
+
+(defmethod current-datetime-base-type :default
+  [_]
+  :type/DateTime)
```

Comment:
1. Seeing how this is only used in tests, it should be a test method rather than a driver method itself. 3rd party driver authors or future driver authors at MB are going to see this and not be sure if they need to implement it or why they'd want to do so
2. Why not just update the dox for `current-datetime-honeysql-form` and say that it should preferably return a HoneySQL form with type information? And have `:presto-jdbc` do this. This is arguably something they should already be doing anyway, for the same reason all the other temporal-related stuff should be doing it -- so we can avoid unneeded casts/use them when needed/use appropriate functions for databases that don't allow implicit conversions between different temporal types.  If the form has type info, the tests could convert the database type to a metabase base type using `database-type->metabase-type`, otherwise fall back to `:type/DateTime`
3. New driver multimethods, if you do add them, should have `:added "0.40.0"` metadata so we can keep track of when we added stuff so we can inform 3rd-party driver authors

Reply:
Since this information could potentially come back as part of the type information on the HoneySQL form returned by `current-datetime-honeysql-form`, if we _were_ to add this method, we could potentially have conflicts between what the two methods returned. e.g. maybe my driver has

```clj
(defmethod current-datetime-honeysql-form :my-driver
  [_]
  (hx/with-database-type-info :%now "date"))

(defmethod current-datetime-base-type :my-driver
  [_]
  :type/DateTimeWithTZ)
```

That's clearly _wrong_, and it's not clear which one should be considered correct. If we're trying to make sure we cast `current-datetime-honeysql-form` correctly, do we treat it as a `date`, or as whatever `(database-type->honeysql-type :type/DateTimeWithTZ)` returns?

The fact that this method would introduce a second source of truth should be a red flag that we might not actually need it

Reply:
OK, good points.  I got rid of the multimethod and instead just changed `current-datetime-honeysql-form` to attach the database type to the form (for Presto JDBC) and changed the test to check for that.